PSKoans/Koans/Constructs and Patterns/AboutRegularExpressions.Koans.ps1

```diff
@@ -0,0 +1,468 @@
+        the most common and useful special symbols in regex.
+        
+        This is ^\where things st\Art getting \weir\D$
+    #>
+    
+
+    Context '. (period)' {
+        It 'matches any character' {
+            # The . (period) matches literally any character
+    
+            [bool]$morningMatch = $____ # Should be either $true or $false
+            $morningMatch | Should -Be 'Lazy Sunday mornings' -match 'S.nday'
+    
+            '____' -match 'invi.a.ion' | Should -BeTrue
+        }
+    }
+
+    Context '\n' {
+        It 'matches new lines' {
+            <#
+                The \n symbol is the first regex symbol you've encountered that isn't just a single
+                character. Normally if you saw a lowercase "n" in a regex, it would just mean
+                "literally the letter n". When it's preceded by a backslash, however, that "n" takes
+                on a special meaning. In this case, \n matches new lines.
+    
+                There are many more regex symbols that are single letters preceded by a backslash.
+                In fact, the backslash is probably the single most important character in all of
+                regex.
+            #>
+    
+            $multiLine = @"
+They might look
+similar and look
+soft, but you should
+never confuse a
+wild cougar for
+a domestic cat
+"@
+            $catPattern = '____\n'
+            $multiLine -match $catPattern | Should -BeTrue
+        }
+    }
+
+    Context '\d and \D' {
+        It 'matches digits and non-digits' {
+            <#
+                The \d symbol matches digits (the numbers 0 - 9 inclusive), while the symbol \D
+                matches anything OTHER THAN a digit. Yes, regular expressions are case sensitive.
+                This "lower case for affirmative, upper case for negative" pattern is common among
+                regex symbols as you'll discover as you carry on.
+            #>
+    
+            '__' -match '\d' | Should -BeTrue
+            '__' -match '\d\d\d' | Should -BeTrue
+            '__' -match '\D' | Should -BeTrue
+        }
+    }
+
+    Context '\w and \W' {
+        It 'matches word characters' {
+            <#
+                \w matches word characters which are letters, numbers, and underscores. The \W
+                symbol matches anything else.
+            #>
+            '***' -match '____' | Should -BeTrue
+            'warmth' -match '____' | Should -BeTrue
+        }
+    }
+
+    Context '^ (caret) and $ (dollar sign)' {
+        It 'matches the start and end of lines' {
+            <#
+                The ^ symbol matches the start of a line, which is not normally represented in
+                English with a character. The $ represents the end of a line.
+            #>
+    
+            $startAndEnd = '____'
+            $startAndEnd -match '^a' | Should -BeTrue
+            $startAndEnd -match 'z$' | Should -BeTrue
```

Comment:
Separate `It` blocks here too, which would probably remove the need for the above comment since the `It` names would probably provide that information.